haplotypecaller-gvcf-gatk4.wdl

```diff
@@ -88,77 +87,95 @@ workflow HaplotypeCallerGvcf_GATK4 {
 task HaplotypeCaller {
   File input_bam
   File input_bam_index
-  String gvcf_name
+  File interval_list
+  String vcf_basename
   File ref_dict
   File ref_fasta
   File ref_fasta_index
-  File interval_list
-  Int? interval_padding
   Float? contamination
-  Int? max_alt_alleles
+  Boolean make_gvcf
 
-  Int preemptible_tries
-  Int disk_size
-  String mem_size
-
-  String docker_image
   String gatk_path
-  String java_opt
-
-  command {
-    ${gatk_path} --java-options ${java_opt} \
+  String? java_options
+  String java_opt = select_first([java_options, "-XX:GCTimeLimit=50 -XX:GCHeapFreeLimit=10"])
+
+  # Runtime parameters
+  String docker
+  Int? mem_gb
+  Int? disk_space_gb
+  Boolean use_ssd = false
+  Int? cpu
+  Int? preemptible_attempts
+
+  Int machine_mem_gb = select_first([mem_gb, 7])
+  Int command_mem_gb = machine_mem_gb - 1
+
+  command <<<
+  set -e
+  
+    ${gatk_path} --java-options "-Xmx${command_mem_gb}G ${java_opt}" \
       HaplotypeCaller \
       -R ${ref_fasta} \
       -I ${input_bam} \
-      -O ${gvcf_name} \
       -L ${interval_list} \
-      -ip ${default=100 interval_padding} \
-      -contamination ${default=0 contamination} \
-      --max-alternate-alleles ${default=3 max_alt_alleles} \
-      -ERC GVCF
-  }
+      -O ${vcf_basename}.vcf.gz \
```

Comment:
I'd like to keep the logic I had to keep the .g.vcf.gz extension if the task is run in GVCF mode.